pulp_ansible/app/tasks/collections.py

```diff
@@ -50,60 +51,66 @@ def nowhere(*args, **kwargs):
+    import_pb = ProgressBar(message='Importing Collections', total=len(repository_spec_strings))
 
     with RepositoryVersion.create(repository) as new_version:
         with tempfile.TemporaryDirectory() as temp_ansible_path:
-
-            # workaround: mazer logs errors without this dir  https://pulp.plan.io/issues/4999
-            os.mkdir(temp_ansible_path + os.sep + 'ansible_collections')
-
-            galaxy_context = GalaxyContext(
-                collections_path=temp_ansible_path,
-                server={
-                    'url': remote.url,
-                    'ignore_certs': False,
-                },
-            )
-
-            install_repository_specs_loop(
-                display_callback=nowhere,
-                galaxy_context=galaxy_context,
-                repository_spec_strings=repository_spec_strings,
-            )
-
-            content_walk_generator = os.walk(temp_ansible_path)
-            for dirpath, dirnames, filenames in content_walk_generator:
-                if 'MANIFEST.json' in filenames:
-                    with open(dirpath + os.path.sep + 'MANIFEST.json') as manifest_file:
-                        manifest_data = json.load(manifest_file)
-                    info = manifest_data['collection_info']
-                    filename = '{namespace}-{name}-{version}'.format(
-                        namespace=info['namespace'],
-                        name=info['name'],
-                        version=info['version'],
-                    )
-                    tarfile_path = temp_ansible_path + os.path.sep + filename + '.tar.gz'
-                    with tarfile.open(name=tarfile_path, mode='w|gz') as newtar:
-                        newtar.add(dirpath, arcname=filename)
-
-                    with transaction.atomic():
-                        collection, created = Collection.objects.get_or_create(
+            with download_pb:
+                # workaround: mazer logs errors without this dir https://pulp.plan.io/issues/4999
+                os.mkdir(temp_ansible_path + os.sep + 'ansible_collections')
+
+                galaxy_context = GalaxyContext(
+                    collections_path=temp_ansible_path,
+                    server={
+                        'url': remote.url,
+                        'ignore_certs': False,
+                    },
+                )
+
+                install_repository_specs_loop(
+                    display_callback=nowhere,
+                    galaxy_context=galaxy_context,
+                    repository_spec_strings=repository_spec_strings,
+                )
+
+                download_pb.done = len(repository_spec_strings)
+
+            with import_pb:
+                content_walk_generator = os.walk(temp_ansible_path)
+                for dirpath, dirnames, filenames in content_walk_generator:
+                    if 'MANIFEST.json' in filenames:
+                        with open(dirpath + os.path.sep + 'MANIFEST.json') as manifest_file:
```

Comment:
Normally I see `os.path.join`. Curious why you're doing it this way?

Reply:
Not for any particular reason. I should probably switch to `os.path.join` which I believe will have better performance. I'll do that now.

Reply:
pushed